CDデプロイの修正

## 概要
<!--
  この機能追加の目的・背景・解決したい課題を2〜3行で箇条書きにする。
  例:
  - ユーザーが〇〇できないため、△△が不便な状態
  - 〇〇機能を追加することで△△を解決する
-->
```
Run php artisan ziggy:generate
PHP Warning:  require(/home/runner/work/keiba-db-gen2/keiba-db-gen2/source/vendor/autoload.php): Failed to open stream: No such file or directory in /home/runner/work/keiba-db-gen2/keiba-db-gen2/source/artisan on line 10
PHP Stack trace:
PHP   1. {main}() /home/runner/work/keiba-db-gen2/keiba-db-gen2/source/artisan:0
PHP Fatal error:  Uncaught Error: Failed opening required '/home/runner/work/keiba-db-gen2/keiba-db-gen2/source/vendor/autoload.php' (include_path='.:/usr/share/php') in /home/runner/work/keiba-db-gen2/keiba-db-gen2/source/artisan:10
Stack trace:
#0 {main}
  thrown in /home/runner/work/keiba-db-gen2/keiba-db-gen2/source/artisan on line 10
Error: Process completed with exit code 255.

```

## 機能要件
<!--
  実装すべき具体的な振る舞いをユーザー視点で箇条書きにする。
  「〜できる」「〜が表示される」など動詞を使って記述する。
  例:
  - ユーザーは〇〇画面で△△ボタンを押すと〜できる
  - システムは〇〇の場合に△△を返す
-->

## 制約
<!--
  実装時に守るべき非機能要件・技術的制約・スコープ外を箇条書きにする。
  例:
  - 既存の〇〇APIの仕様を変更しない
  - パフォーマンス要件: レスポンス〇〇ms以内
  - 今回のスコープ外: △△は別issueで対応
-->

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -11,7 +11,7 @@ jobs:
 
     defaults:
       run:
-          working-directory: source
+        working-directory: source
 
     env:
       SSH_PRIVATE_KEY: ${{ secrets.SECRET_KEY }}
@@ -25,30 +25,26 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      # ② Node セットアップ
+      # ② Nodeセットアップ
       - name: Setup Node
         uses: actions/setup-node@v4
         with:
-          node-version: 24
-          cache: 'npm'
+          node-version: 20.19
+          cache: 'pnpm'
           cache-dependency-path: source/pnpm-lock.yaml
 
-      - name: enable pnpm
+      - name: Enable pnpm
         run: corepack enable
 
-      # ③ フロントビルド
+      # ③ フロント依存インストール
       - name: Install dependencies
         run: pnpm install
 
-      - name: Generate frontend resources
-        run: |
-          php artisan ziggy:generate
-          php artisan route:cache
-
+      # ④ フロントビルド（★CIではここまで）
       - name: Build assets
         run: pnpm run build
 
-      # ④ SSH セットアップ（安全版）
+      # ⑤ SSH セットアップ
       - name: Setup SSH
         run: |
           set -e
@@ -61,7 +57,7 @@ jobs:
           ssh-keyscan -H "$HOST" >> ~/.ssh/known_hosts
           chmod 644 ~/.ssh/known_hosts
 
-      # ⑤ アセット転送（差分・安全）
+      # ⑥ アセット転送
       - name: Deploy assets via rsync
         run: |
           rsync -avz --delete \
@@ -69,19 +65,18 @@ jobs:
             ./public/build/ \
             ${USER}@${HOST}:${ASSETS_BUILD_FILE_PATH}
 
-      # ⑥ サーバー側デプロイ（本体）
+      # ⑦ サーバー側デプロイ（Laravel全部ここ）
       - name: Run remote deploy
         run: |
-          ssh ${USER}@${HOST} << EOF
+          ssh ${USER}@${HOST} << 'EOF'
             set -e
 
-            echo "== Move to project directory =="
             cd ${LARAVEL_DIR}
 
             echo "== Enable maintenance mode =="
             php artisan down || true
 
-            echo "== Git update =="
+            echo "== Git pull =="
             git fetch origin
             git reset --hard origin/main
 
@@ -92,22 +87,23 @@ jobs:
               --optimize-autoloader \
               --no-interaction
 
+            echo "== Generate Laravel assets =="
+            php artisan ziggy:generate || true
+            php artisan route:cache
+            php artisan config:cache
+            php artisan view:cache
+
             echo "== Run migrations =="
             php artisan migrate --force
 
-            echo "== Clear cache =="
-            php artisan cache:clear
-            php artisan config:clear
-            php artisan route:clear
-            php artisan view:clear
+            echo "== Clear caches =="
+            php artisan cache:clear || true
+            php artisan config:clear || true
+            php artisan route:clear || true
+            php artisan view:clear || true
 
-            echo "== Rebuild cache =="
-            php artisan config:cache
-            php artisan route:cache
-            php artisan view:cache
-
-            echo "== Disable maintenance mode =="
+            echo "== Enable app =="
             php artisan up
 
-            echo "== Deploy completed successfully =="
+            echo "== Deploy completed =="
           EOF

